diff --git a/Makefile b/Makefile
index 21b82d9..b96c52a 100644
--- a/Makefile
+++ b/Makefile
@@ -12,13 +12,13 @@ MAN5 = farbfeld.5
 all: ${BIN}
 
 png2ff ff2png:
-	${CC} -o $@ ${CFLAGS} ${CPPFLAGS} -L${PNGLIB} -lpng -I${PNGINC} ${LDFLAGS} $@.c
+	${CC} -o $@ ${CFLAGS} ${CPPFLAGS} -L${PNGLIB} -I${PNGINC} ${LDFLAGS} $@.c -lpng
 
 jpg2ff ff2jpg:
-	${CC} -o $@ ${CFLAGS} ${CPPFLAGS} -L${JPGLIB} -ljpeg -I${JPGINC} ${LDFLAGS} $@.c
+	${CC} -o $@ ${CFLAGS} ${CPPFLAGS} -L${JPGLIB} -I${JPGINC} ${LDFLAGS} $@.c -ljpeg
 
 .c:
-	${CC} -o $@ ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} $<
+	${CC} $< -o $@ ${CFLAGS} ${CPPFLAGS} ${LDFLAGS}
 
 clean:
 	rm -f ${BIN}
diff --git a/config.mk b/config.mk
index 4ecba54..8ab744e 100644
--- a/config.mk
+++ b/config.mk
@@ -8,10 +8,10 @@ PREFIX = /usr/local
 MANPREFIX = ${PREFIX}/man
 
 PNGLIB = /usr/local/lib
-PNGINC = /usr/local/include
+PNGINC = /usr/include/libpng12
 
 JPGLIB = /usr/local/lib
-JPGINC = /usr/local/include
+JPGINC = /usr/lib/i386-linux-gnu
 
 INCS =
 LIBS =
